blacktie/tx/src/test/cpp/TestTransactions.cxx

```diff
@@ -527,11 +527,14 @@ void TestTransactions::test_wait_for_recovery()
 	(void) dummy_rm_add_fault(fault2);
 	(void) dummy_rm_add_fault(fault3);
 	BT_ASSERT_EQUAL(TX_OK, tx_open());
-	while ((rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1) && nsecs != 0) {
-		if (nsecs-- % 10 == 0) {
-			btlogger("TestTransactions::test_run_recovery sleeping for %d seconds", nsecs);
+	nrecs = count_log_records();
+	if(nrecs > 0 ) {
+		while ((rcCnt2 == 0 || (nrecs = count_log_records()) == nrecs1) && nsecs != 0) {
+			if (nsecs-- % 10 == 0) {
+				btlogger("TestTransactions::test_run_recovery sleeping for %d seconds", nsecs);
+			}
+			doSix(1);
```

Comment:
The previous test makes no sense unless you move the sleep (doSix) inside the if block